x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMStatDisruptionIT.java

```diff
@@ -0,0 +1,564 @@
+        .put(LeaderChecker.LEADER_CHECK_TIMEOUT_SETTING.getKey(), "5s") // for hitting simulated network failures quickly
+        .put(LeaderChecker.LEADER_CHECK_RETRY_COUNT_SETTING.getKey(), 1) // for hitting simulated network failures quickly
+        .put(FollowersChecker.FOLLOWER_CHECK_TIMEOUT_SETTING.getKey(), "5s") // for hitting simulated network failures quickly
+        .put(FollowersChecker.FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), 1) // for hitting simulated network failures quickly
+        .put(Coordinator.PUBLISH_TIMEOUT_SETTING.getKey(), "5s") // <-- for hitting simulated network failures quickly
+        .put(LagDetector.CLUSTER_FOLLOWER_LAG_TIMEOUT_SETTING.getKey(), "5s") // remove lagging nodes quickly so they can rejoin
+        .put(TransportSettings.CONNECT_TIMEOUT.getKey(), "10s") // Network delay disruption waits for the min between this
+        // value and the time of disruption and does not recover immediately
+        // when disruption is stop. We should make sure we recover faster
+        // then the default of 30s, causing ensureGreen and friends to time out
+        .build();
+
+    /**
+     * Test that after successful snapshot preRegisteredRuns status is 0.
+     */
+    public void testSuccessSnapshot() throws Exception {
+        final String idxName = "test-idx";
+        final String repoName = "test-repo";
+        final String policyName = "test-policy";
+
+        internalCluster().startMasterOnlyNodes(1);
+        final String masterNode = internalCluster().getMasterName();
+        final String dataNode = internalCluster().startDataOnlyNode();
+        ensureStableCluster(2);
+
+        createRandomIndex(idxName, dataNode);
+        createRepository(repoName, "mock");
+        createSnapshotPolicy(policyName, "snap", NEVER_EXECUTE_CRON_SCHEDULE, repoName, idxName);
+
+        ensureGreen();
+
+        String snapshotName = executePolicy(masterNode, policyName);
+        logger.info("Created snapshot: " + snapshotName);
+
+        waitForSnapshot(repoName, snapshotName);
+
+        assertBusy(() -> {
+            assertSnapshotSuccess("test-repo", snapshotName);
+            logger.info("Verified that snapshot was not successful");
+        }, 1, TimeUnit.MINUTES);
+
+        assertBusy(() -> {
+            var snapshotLifecycleMetadata = getSnapshotLifecycleMetadata();
+            var snapshotLifecyclePolicyMetadata = snapshotLifecycleMetadata.getSnapshotConfigurations().get(policyName);
+            assertStats(snapshotLifecycleMetadata, policyName, 1, 0);
+            assertNull(snapshotLifecyclePolicyMetadata.getLastFailure());
+            assertNotNull(snapshotLifecyclePolicyMetadata.getLastSuccess());
+            assertEquals(0, snapshotLifecyclePolicyMetadata.getInvocationsSinceLastSuccess());
+            assertEquals(Set.of(), snapshotLifecyclePolicyMetadata.getPreRegisteredSnapshots());
+        }, 1, TimeUnit.MINUTES);
+    }
```

Comment:
I've been unable to build a test that checks the situation where a  snapshot is currently running and another fails. To do this, I need a way to delay the first snapshot so that it is not un-registered until after the second snapshots completes. I haven't figured out a way to do this; any ideas?